Editor/Brushes/PrefabBrushes/PrefabBrush/PrefabBrush.cs

```diff
@@ -46,7 +46,8 @@ public override void Paint(GridLayout grid, GameObject brushTarget, Vector3Int p
         {
             var objectsInCell = GetObjectsInCell(grid, brushTarget.transform, position);
             var existPrefabObjectInCell = objectsInCell.Any(objectInCell => PrefabUtility.GetCorrespondingObjectFromSource(objectInCell) == m_Prefab);
-
+            m_Rotation = Quaternion.Euler(m_Prefab.transform.eulerAngles);
```

Comment:
m_Rotation stores the rotation of the brush which can be changed using the rotation hot-keys. This will always override this stored rotation.

We could keep both rotations with the following code:

```suggestion
            m_Rotation = Quaternion.Euler(m_Prefab.transform.eulerAngles);
```
```suggestion
            var rotation = Quaternion.Euler(m_Prefab.transform.eulerAngles) * m_Rotation;
            
            if (!existPrefabObjectInCell)
            {
                base.InstantiatePrefabInCell(grid, brushTarget, position, m_Prefab, rotation);
            }
```

Would this be better and still suit your use-case?